src/shared_modules/content_manager/README.md

```diff
@@ -0,0 +1,341 @@
+# Content Manager
+
+The Content Manager is a module that is in charge of obtaining the Wazuh Content from an external repository, as well as maintaining the periodic updates that the manager will obtain.
+
+## Usage
+
+The Content Manager counts with a [test tool](./testtool/main.cpp) that can be used to perform tests, try out different configurations, and to better understand the module.
+
+The input configuration of the Content Manager is described below:
+
+- `topicName`: Topic name, used to represent the actions being executed.
+- `interval`: Interval, in seconds, between each action execution.
+- `ondemand`: If true, the module will be executed on demand.
+- `configData`: Configuration data to create the orchestration of the module.
+  + `contentSource`: Source of the content. Can be any of `api`, `cti-api`, `file`, or `offline`.
+  + `compressionType`: Compression type of the content. Can be any of `gzip`, `zip`, or `xz`.
+  + `versionedContent`: Type of versioned content. Can be any of `false` (content versioning disabled) or `cti-api`.
```

Comment:
Yes... and no :sweat_smile:

If we enable the `cti-api` content versioner with a downloader different from `cti-api`, the version updater will be instantiated and will update the database offset (offset that shouldn't have changed since there is no `cti-api` downloader).

I will add a comment to clarify this.

Reply:
Done.

Reply:
> If we enable the cti-api content versioner with a downloader different from cti-api, the version updater will be instantiated and will update the database offset (offset that shouldn't have changed since there is no cti-api downloader).

But is that a problem? I mean, setting a non-cti-api downloader and setting versionedContent with cti-api: is it a config error? should it be a config error? will something break?

Reply:
It's not a config error, nor will end up breaking anything.
It will just execute a version updater that won't have any purpose (updating an offset that didn't change).

Should it be a config error? Not that I know of. Maybe @Damian-Mangold can tell.